Lib/wsgiref/simple_server.py

```diff
@@ -71,6 +71,12 @@ def set_app(self,application):
 class WSGIRequestHandler(BaseHTTPRequestHandler):
 
     server_version = "WSGIServer/" + __version__
+    server_handler = ServerHandler
+
+    def __init__(self, *args, **kwargs):
+        if not issubclass(self.server_handler, BaseHandler):
```

Comment:
I don't think this check is appropriate, if someone duck-types `BaseHandler`, the RequestHandler should accept it restricting the user.

Reply:
Why would someone duck-type `BaseHandler` instead of just subclassing it. If there's a good reason then that is fine, otherwise we are just supporting calling this in yet another way, and that way requires us to implement a more complex verification routine.

Reply:
@remilapeyre I agree with @FFY00, we should make it as simple as we can

Reply:
I don't think check a class variable inside instance constructor is a good idea. I think it's fine to omit it and use `server_handler` directly. Those who want to extend should ensure it's of correct type.